src/towncrier/test/test_build.py

```diff
@@ -836,3 +844,65 @@ def test_start_string(self):
         """)
 
         self.assertEqual(expected_output, output)
+
+    def test_with_topline_and_template_and_draft(self):
+        """
+        Spacing is proper when drafting with a topline and a template.
```

Comment:
I am not sure what the difference between `test_title_format_custom` and this test.

Reply:
It at least differs in that this specifies a custom template while `test_title_format_custom` leaves the default template in use.  When writing this test it ended up a bit confusing to me too how all the bits were interacting but it did seem to recreate the issue reported at #105.

I feel like the existing implementation code had evolved and needed cleanup but as it was it offered corners where specifying a title format \_with\_ a custom template \_and\_ draft (and maybe something about the title-less feature section?) could result in some unique bug.

Reply:
Understood. But I think that this test can be removed.
Now that we use the exact same output for draft and final,#105 should be fixed.

I would prefer not to have this tests as it's long and confusing.

The test only checks the output of draft.

A good test for #105 would had generated the output with and without draft and compare them to make sure they are exactly the same.

But I don't think that we need extra test for that.